js/ding_place2book.js

```diff
@@ -3,11 +3,23 @@
  *
  * Provide event nodes/pages with ticket info from Place2book
  */
-jQuery(document).ready(function($) {
-	$('.place2book-ticketinfo').each(function() { 
-	    var obj = this;
-		$.getJSON(Drupal.settings.basePath + 'ding/place2book/ticketinfo/' + this.value, function(data) {
-		  $(obj).replaceWith(data['markup']);
-		});		
-    });
-});
\ No newline at end of file
+(function ($) {
+
+  Drupal.behaviors.ding_place2book = {
+    attach: function (context, settings) {
+      $('.place2book-ticketinfo').once('place2book-button', function() {
+        var request = $.ajax({
+          url: Drupal.settings.basePath + 'ding/place2book/ticketinfo/' + this.value,
+          type: 'POST',
+          dataType: 'json',
+          success: ding_place2book_insert,
+        });
+      });
+    }
+  };
+
+  var ding_place2book_insert = function(ding_place2book) {
+    $('.place2book-ticketinfo').replaceWith(ding_place2book.markup);
```

Comment:
As this is used as a callback from the Ajax call it and not using "this" inside to get the element it will hit all '.place2book-ticketinfo' on the page if use on pages with more than one ticket.

Before it used var obj = this....